src/components/footer/footer-mapper.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { NavLink } from "react-router-dom";
+import "./footer.scss";
+import {
+  ALL_PATH,
+  ALL_STYLE_ICON,
+  ALL_LINK,
+  FOOTER_LABELS,
+} from "../../constants/footerConstants";
+
+const { HOME, CREATE_EVENT, SEARCH_EVENT, TERMS, ABOUT } = FOOTER_LABELS;
+const { LINK_FACEBOOK, LINK_INSTAGRAM, LINK_YOUTUBE } = ALL_LINK;
+const {
+  PATH_HOME,
+  PATH_CREATE_EVENT,
+  PATH_SEARCH_EVENT,
+  PATH_TERMS,
+  PATH_ABOUT,
+} = ALL_PATH;
+
+const { STYLE_ICON_FACEBOOK, STYLE_ICON_INSTAGRAM, STYLE_ICON_YOUTUBE } =
+  ALL_STYLE_ICON;
+
+const LiNKS = [
+  { path: PATH_HOME, name: HOME },
+  { path: PATH_SEARCH_EVENT, name: SEARCH_EVENT },
+  { path: PATH_CREATE_EVENT, name: CREATE_EVENT },
+  { path: PATH_TERMS, name: TERMS },
+  { path: PATH_ABOUT, name: ABOUT },
+];
+
+const SOCIAL_LINKS = [
+  { icon: STYLE_ICON_FACEBOOK, link: LINK_FACEBOOK },
+  { icon: STYLE_ICON_INSTAGRAM, link: LINK_INSTAGRAM },
+  { icon: STYLE_ICON_YOUTUBE, link: LINK_YOUTUBE },
+];
+
+export const MapperLink = classLink => {
```

Comment:
Please rename it to `NAV_LINKS`

And I meant before to make it as constant variable, not a function.
Something like `const NAV_LINKS = LINKS.map(...)`. 
It will be created one time here, and then we can just to use it component piece inside main component, and it will not be recreated each rerender

Reply:
I did it for const SOCIAL_NAV_LINKS =     too